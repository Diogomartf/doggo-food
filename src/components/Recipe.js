import React from 'react';
import styled from 'styled-components';

const Recipe = ({ title, image, url, ingredients }) => {
  const hasLactose =
    ingredients.includes('cheese') || ingredients.includes('milk');

  return (
    <RecipeBox href={url}>
      <ImageBackground src={image}>
        {hasLactose && <Label>Lactose</Label>}
      </ImageBackground>

      <Content>
        <Title>{title}</Title>
      </Content>
      <Ingredients>{ingredients}</Ingredients>
    </RecipeBox>
  );
};

const RecipeBox = styled.a`
  text-decoration: none;
  outline: none;
  cursor: pointer;
`;

const Label = styled.div`
  position: absolute;
  top: 20px;
  right: 0;
  background: ${props => props.theme.colors.primary};
  color: ${props => props.theme.colors.white};

  transform: rotate(45deg);
`;
Label.displayName = 'Label';

const Content = styled.div`
  margin-top: 12px;
`;

const Ingredients = styled.div`
  font-weight: 600;
  font-size: 12px;
  text-transform: uppercase;

  color: ${({ theme }) => theme.colors.darkGray};
`;

const Title = styled.div`
  margin-top: 12px;
  margin-bottom: 4px;

  font-size: 18px;
  font-weight: 600;
  color: ${({ theme }) => theme.colors.black};
`;

const ImageBackground = styled.div`
  position: relative;
  width: 100%;
  height: 262px;
  background-image: url("${({ src }) => src}");

  background-color: ${({ theme }) => theme.colors.gray};
  background-size: cover;
  background-position: 50% 50%;
  background-repeat: no-repeat;
`;

export default Recipe;
