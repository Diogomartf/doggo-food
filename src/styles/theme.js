import colors from './colors';

export default {
  name: 'default theme',
  colors
};
