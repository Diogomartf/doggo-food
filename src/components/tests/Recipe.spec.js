import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles';

import Recipe from '../Recipe';

const recipe = {
  title: 'Alfredo Recipe',
  href: 'http://www.grouprecipes.com/77213/alfredo-sauce-weight-watchers.html',
  ingredients: 'garlic, pesto, water',
  thumbnail: 'http://img.recipepuppy.com/306873.jpg'
};

describe('<Recipe/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Recipe
        key={recipe.href}
        title={recipe.title}
        ingredients={recipe.ingredients}
        url={recipe.href}
        image={recipe.thumbnail}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render label when no lactose ingredients', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Recipe
          key={recipe.href}
          title={recipe.title}
          ingredients={recipe.ingredients}
          url={recipe.href}
          image={recipe.thumbnail}
        />
      </ThemeProvider>
    );

    expect(wrapper.find('Label')).toHaveLength(0);
  });

  it('should render label when cheese in ingredients', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Recipe
          key={recipe.href}
          title={recipe.title}
          ingredients="garlic, pesto, cheese"
          url={recipe.href}
          image={recipe.thumbnail}
        />
      </ThemeProvider>
    );

    expect(wrapper.find('Label')).toHaveLength(1);
  });

  it('should render label when milk in ingredients', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Recipe
          key={recipe.href}
          title={recipe.title}
          ingredients="garlic, pesto, milk"
          url={recipe.href}
          image={recipe.thumbnail}
        />
      </ThemeProvider>
    );

    expect(wrapper.find('Label')).toHaveLength(1);
  });
});
