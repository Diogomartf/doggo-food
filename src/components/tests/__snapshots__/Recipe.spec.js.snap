// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe/> should render correctly 1`] = `
<styled.a
  href="http://www.grouprecipes.com/77213/alfredo-sauce-weight-watchers.html"
>
  <styled.div
    src="http://img.recipepuppy.com/306873.jpg"
  />
  <styled.div>
    <styled.div>
      Alfredo Recipe
    </styled.div>
  </styled.div>
  <styled.div>
    garlic, pesto, water
  </styled.div>
</styled.a>
`;
